Add "feed.provider" to the DHO

sykaeh opened the pull request:
This implements #725 (also mentioned in #733) and includes some typo fixes.

This only adds `feed.provider` to the DHO and implements adding it to the report if the parameter `provider` is set in a collector (analogous to `feed.code`). Whether this value is set in the parsers is still open for discussion.

sykaeh commented:
Does anyone understand why the cymru whois expert bot test failed on Python3.4?

dmth commented:
Thank you for implementing this! 

> Does anyone understand why the cymru whois expert bot test failed on Python3.4?

AFAIK the tests make a real lookup at team cymru. If the response differs, the test fails. I don't know why especially the 3.4 Test fails. Maybe there was a short Hickup?

aaronkaplan commented:
Hi @sykaeh ,

thanks for your submission.
It seems the cymru whois unit test simply failed since something changed in the output format there:
```
-  'destination.as_name': 'Local Datacenter Soluções em Comunicação Ltda.,,, BR',
?                                                                         --
+  'destination.as_name': 'Local Datacenter Soluções em Comunicação Ltda., BR',
```

So, I'll fix that test and we'll be fine again and re-run the tests.
Thanks
a

sykaeh commented:
Previously only one of the three Python versions failed. It seems like Cymru is not always returning the same value. That's why I had changed it. I now changed it back. We'll see if it works.

aaronkaplan commented:
@sykaeh okay, got it. Maybe we then have to re-write the test. Anyway. Thanks! It seems to have done the trick for now. Merging this PR.
